src/emu/debug/debugcpu.cpp

```diff
@@ -1008,6 +1007,37 @@ u64 debugger_cpu::expression_read_memory(void *param, const char *name, expressi
 			break;
 		}
 
+		case EXPSPACE_OPCODE:
+		{
+			device_t *device = nullptr;
+			device_memory_interface *memory;
+
+			if (name != nullptr)
+				device = expression_get_device(name);
+			if (device == nullptr || !device->interface(memory))
+			{
+				device = get_visible_cpu();
+				memory = &device->memory();
+			}
+			if (memory->has_space(AS_DECRYPTED_OPCODES + (spacenum - EXPSPACE_OPCODE)))
+			{
+				if (!with_se) {
+			  		auto dis = m_machine.disable_side_effect();
+			  		return expression_read_program_direct(memory->space(AS_DECRYPTED_OPCODES), (spacenum == EXPSPACE_OPCODE), address, size);
```

Comment:
Same here.  Plus, in practice, the second parameter is unused, and the read path is broken.  Now that I read it more closely, this code is horrible (but not your fault).